subsys/net/ip/net_mgmt.c

```diff
@@ -20,6 +20,14 @@
 struct mgmt_event_entry {
 	u32_t event;
 	struct net_if *iface;
+
+#ifdef CONFIG_NET_MGMT_EVENT_INFO
+	union {
+		u8_t info[CONFIG_NET_MGMT_EVENT_INFO_SIZE];
```

Comment:
This is very unfortunate, as it bloats the event entry. I wonder if we could allocate the space outside of mgmt_event_entry and only store pointer here.

Reply:
I believe the cost of code to manage external memory placeholder vs the numbers of entries vs the number of event that do not provide info makes it ok as it is now.

As far as I can tell, the info placeholder won't be that big. Currently I set it to 8bytes, but I think it will be more like 10-12 bytes. (ipv6 address + prefix for routes for instance?) it's really not much. Multiply that with number of event entries (2 by default btw) it's quite a limited amount of memory in the end, and from code perspective it was a minor amount of lines added as you can see..

With external placeholder you would still need to be prepared that all the event queued entries provide info, i.e.: you would need as many placeholders as event entries. Now, add to that it could be those would be of the maximum size info (let's say 4 routes added one by one) so you need to be prepared to provide max size for all. Now add also the pointer to store the placeholder on the entry (4bytes more). In the end, with current use case, it would be eating more memory.

Reply:
Lets see how it goes, the amount of data will be much higher as IPv6 address is 16 bytes, and routes are typically 16+1.

Reply:
indeed 16bytes, but that's still quite limited imo.

Reply:
As said in my previous reply: with internal or external memory allocation we have to make sure the worst case works. Worst case being: all queued events use the biggest size of info.

So in the end the memory allocation is the same (we do not share our reserved pools with anything else), being in fact a bit worse for external method since each entry needs an attributes pointing to the allocated memory.

I don't think we'll ever get huge amount of data to pass, nor huge size of entry queue in net_mgmt, so in the end it's just fine. I mean, we are talking of entry made of 4+4 bytes currently + 4 + probably 20 (16+1, with alignment it will be 20): 32 bytes. It's nothing compared to the current net buffer which size is a minimal of 128bytes. And we are not supposed to have a large amount of entries also.

Plus, we can play on the mgmt thread priority to un-queue events faster if required, to lower the need of larger queue. In any case, about this event queue size: net mgmt is not meant to be bullet proof guarantied on message passing. If the system is over-loaded, on a full event queue: the oldest will get scraped by a new one. A bit like netlink in a way, if the system is overloaded, the kernel might scrap old message in the socket buffer. There isn't much we can do actually.

Anyway, if really necessary, once Johan has done the split of net_buf metadata/data, we may use the memory pool of net_buf's data as an external allocator. That could be a solution if we really need to pass much bigger set of data as "event information".